Lesson 3

olexandr-gorkavenko opened the pull request:
Добавил решения всех заданий.

olexandr-gorkavenko committed:
Добавил .gitignore

olexandr-gorkavenko committed:
Добавил index.html

olexandr-gorkavenko committed:
Добавил решение 1 задания task1.js

olexandr-gorkavenko committed:
Добавил решение 2 задания task2.js

olexandr-gorkavenko committed:
Добавил решение 3 задания task3.js

olexandr-gorkavenko committed:
Добавил решение 4 задания task4.js

vadimk7 commented on lesson-3/JS/task1.js:
```diff
@@ -0,0 +1,22 @@
+* [1,2,3,4,5] => [1, 3, 6, 10, 15]
+*/
+
+let getArrayOfPartialSums = arr => {
```
почитай, пожалуйста, когда лучше использовать **let/const**

olexandr-gorkavenko replied:
Пока что не получается.

olexandr-gorkavenko replied:
Исправил.

vadimk7 commented on lesson-3/JS/task1.js:
```diff
@@ -0,0 +1,22 @@
+let getArrayOfPartialSums = arr => {
+let result = [];
+
+let sum = arr.reduce((total, amount) => {
```
попробуй реализовать через map + reduce без дополнительного массива

olexandr-gorkavenko replied:
Пока что не получается.

vadimk7 commented on lesson-3/JS/task2.js:
```diff
@@ -0,0 +1,34 @@
+*/
+
+let getArrayAveragePrice = arr => {
+	let arrayAverage = arr.map(item => {
```
переменная arrayAverage не нужна, можно сразу вернуть массив

olexandr-gorkavenko replied:
Исправил.

vadimk7 commented on lesson-3/JS/task3.js:
```diff
@@ -0,0 +1,24 @@
+
+let getArrayFilteredByPrice = arr => {
+	let arrFiltered = arr.filter(item => {	
+		return item.averagePrice > 5000;
```
return и стрелочная функция с 1 строкой - не хорошо

olexandr-gorkavenko replied:
Исправил, спасибо за подсказку.

vadimk7 commented on lesson-3/JS/task3.js:
```diff
@@ -0,0 +1,24 @@
+		return item.averagePrice > 5000;
+		});
+	
+	return arrFiltered.sort(compareNumeric);
```
лучше чейнить операторы 

```
arr.filter(...)
   .sort(...)
```

olexandr-gorkavenko replied:
Сделано.

vadimk7 commented on lesson-3/JS/task4.js:
```diff
@@ -0,0 +1,116 @@
+/*
+* Не понял суть задания. поэтому 3 варианта решения.
+*
+* В первом и втором варианте просто покажет что expireDate массив объектов, 
```
* expireDate - не массив объектов, а обычный объект
* но с этим массивом дальше работать нельзя - не понял этого, почему нельзя?
* конкатенация значений year и month - нет, это не конкатенация, это ты создал 2 переменные

olexandr-gorkavenko replied:
Почему при записи
console.log(result.cards.expireDate = ${expireDate}); // result.cards.expireDate = [object Object]
а если console.log(expireDate); // {year: 2021, month: 9}

olexandr-gorkavenko committed:
Исправил let/const согласно замечанию;
избавился от переменной arrayAverage в файле task2;
использовал чейнинг в task3;
убрал комментарии в task4.

olexandr-gorkavenko commented:
Исправил задания согласно сделанным замечаниям. Остался неясным вопрос:

> Код в шаблонной подстановке может быть любым выражением JavaScript, так что вызовы функций, арифметика и т.п. разрешены.
> 
> Если какое-то значение не строкового типа, оно будет приведено к строке при помощи обычных правил. 
> К примеру, если это объект, у него вызовется метод .toString().

const {result: [ {cards: [ {expireDate} ] } ] } = jso;

Почему при записи
console.log(`result.cards.expireDate = ${expireDate}`); // result.cards.expireDate = [object Object]

а если console.log(expireDate); // {year: 2021, month: 9}

vadimk7 commented:
@olexandr-gorkavenko потому что вот это `${expireDate}` явно приводит объект к строке, вызывается метод toString

olexandr-gorkavenko committed:
В task2 перенес деструктуризацию в параметры.
